src/Commands/MigrateDumpCommand.php

```diff
@@ -116,8 +116,9 @@ private static function mysqlDump(array $db_config, string $schema_sql_path) : i
         // console output with `$this->info` and `->error`.
         passthru(
             $command_prefix
-            . ' --result-file=' . escapeshellarg($schema_sql_path)
-            . ' --no-data',
+            . ' --result-file=' 
+            . escapeshellarg($schema_sql_path)
+            . (config('migration-snapshot.data') ? '' : ' --no-data'),
```

Comment:
Another concern with relying solely on the config is that every `php artisan migrate` will invoke dumping, even when a development or staging DB may have TBs of data. Perhaps the config could instead be a whitelist of those tables to also dump; like `['order_types', 'statuses', …]`

Reply:
Well whitelisting might not work in that case either as we might add a default system user, but then add in a bunch of users from seeders which would get included on the next migration run.

One solution for that would be to only run the dump when a user manually runs `php artisan migrate:dump` or we only dump when migrations are run from an empty database and seeding data hasn't been added yet.

Reply:
> Well whitelisting might not work in that case either as we might add a default system user, but then add in a bunch of users from seeders which would get included on the next migration run.

@tomschlick I don't think the whitelisting has to be perfect. But I do think a table-name based list would work in most cases. It also forces users who do want data to consider those cases. Also, keep in mind that the implicit dumping is already controlled by `environments`. So if a staging environment has a mixture of stock and real data in some tables, or 'development' often has seeds, then users could just remove those from the (implicitly dumped) environments.

Reply:
Another thought is that the data could go to a `database/migrations/data.sql`, or perhaps the whitelist could even be an indexed array like `['users' => [1,2,3], 'statuses', …]`.